File: Creational/Abstract Factory/TypeScript/src/factory/SUVFactory.ts
// src/factory/SUVFactory.ts
import { IAutoBodyFactory } from './IAutoBodyFactory';
import { AutoBodyPart } from '../product/AutoBodyPart';
import { SUVLeftDoor, SUVRightDoor, SUVHood } from '../product/SUVParts';

export class SUVFactory implements IAutoBodyFactory {
    createLeftDoor(): AutoBodyPart {
        return new SUVLeftDoor();
    }

    createRightDoor(): AutoBodyPart {
        return new SUVRightDoor();
    }

    createHood(): AutoBodyPart {
        return new SUVHood();
    }
}

File: Creational/Abstract Factory/Python/factory/suv_factory.py
# factory/suv_factory.py
from factory import IAutoBodyFactory
from product.suv_parts import SUVLeftDoor, SUVRightDoor, SUVHood

class SUVFactory(IAutoBodyFactory):
    def create_left_door(self) -> SUVLeftDoor:
        return SUVLeftDoor()

    def create_right_door(self) -> SUVRightDoor:
        return SUVRightDoor()

    def create_hood(self) -> SUVHood:
        return SUVHood()
